pkg/controller/container-runtime-config/helpers.go

```diff
@@ -431,29 +452,31 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 // getValidRegistries gets the insecure and blocked registries in the image spec and validates that the user is not adding
 // the registry being used by the payload to the list of blocked registries.
 // If the user is, we drop that registry and continue with syncing the registries.conf with the other registry options
-func getValidRegistries(clusterVersionStatus *apicfgv1.ClusterVersionStatus, imgSpec *apicfgv1.ImageSpec) ([]string, []string, error) {
+func getValidRegistries(clusterVersionStatus *apicfgv1.ClusterVersionStatus, imgSpec *apicfgv1.ImageSpec) ([]string, []string, []string, error) {
```

Comment:
Absolutely non-blocking: I don’t really think it’s worth the extra code to return unmodified `insecureRegs` / `allowedRegs` from this function; the caller can just as well read them directly from `imgSpec` and avoid the return values / variables. OTOH then we would have to think about naming this function something else, and `blockedRegs` would be treated differently from the others…

Reply:
Fixed up to only validate for blocked registries